src/per/qspi.h

```diff
@@ -166,4 +168,113 @@ class QSPIHandle
 
 } // namespace daisy
 
-#endif
+#else
+
+#include <cstdint>
+#include "../tests/TestIsolator.h"
+
+namespace daisy
+{
+/** This is a dummy implementation for use in tests. 
+ *  In your tests you can use this as a placeholder 
+ *  for the physical volatile memory. 
+ *  This provides a block of memory that can be erased, or written
+ *  to.
+ *  
+ *  On the hardware, the memory can be read directly from addresses
+ *  0x90000000 and above. However, within the unit tests you must use
+ *  the Read() function.
```

Comment:
I know this is not actually part of this PR, but maybe it's worth considering for a followup PR

Reply:
If we want to enable the `reinterpret_cast<MyType*>(start_address_)` logic (which can be quite nice!) we could also do it differently: Instead of providing a `Read` function, we could provide a `GetData()` function that returns a pointer to the base memory address.

Reply:
The idea of a `Read()` function is nice, but I prefer the `GetData()` idea instead. That simplifies the actions needed here a bit.

That said, once we get through a few more of these PRs I want to go in and add non-blocking write functions that use the DMA. At that point, I could certainly see the benefits of adding Read function. It could be useful for bigger chunks of data (thinking wavetables, etc.)  to somewhere with faster access times prior to usage, as well as for profiling the memory itself. Which is a little more opaque when memory mapped.

I'll give adding the GetData function a go, and see how it looks.